src/Features/CSharp/Portable/StringIndentation/CSharpStringIndentationService.cs

```diff
@@ -31,6 +31,11 @@ public CSharpStringIndentationService()
             Document document, TextSpan textSpan, CancellationToken cancellationToken)
         {
             var text = await document.GetTextAsync(cancellationToken).ConfigureAwait(false);
+            if (text.IndexOf(@"""""""", textSpan.Start, true) < 0)
+            {
+                return ImmutableArray<StringIndentationRegion>.Empty;
+            }
+
             var root = await document.GetRequiredSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
 
             using var _ = ArrayBuilder<StringIndentationRegion>.GetInstance(out var result);
```

Comment:
i'm genuinely quite confused how this can help.  We should already just be walking the span of the document that intersects `textSpan`.  so this should be blazingly fast as it will just avoid hitting any tree not in that practically immediately.  so we should be examining a tiny set of nodes.